### PR TITLE
Prefetch Skeleton Loading on Indexes and Shows

### LINKED ISSUE #4458: Prefetch Skeleton Loading on Indexes and Shows
# Desired Behavior

As soon as #4456 is complete, we can display the permanent navigation items (Settings, Search...) and the top navbar. 

We should now launch the Prefetch Loading with a new Skeleton Loading. The goal of this step is to load the view content and metadata:

### Index

image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=25426-62817&mode=design&t=75yrmyKfqMepmFVt-11

### Show

image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=25429-70482&mode=design&t=75yrmyKfqMepmFVt-11

### PR DESCRIPTION
**Pull request by gitstart-twenty**
### Description
Prefetch Skeleton Loading on Indexes and Shows

### Refs
#4458

### Demo

https://jam.dev/c/a1ad04e1-80b6-4b2a-b7df-373f52f4b169

https://jam.dev/c/c5038b97-2f18-4c29-8dee-18c09376e5ee

Fixes: #4458

### CODE REVIEW

**Comment by charlesBochet** [comment]
@gitstart-twenty tests are failing :)

File: packages/twenty-front/src/modules/object-metadata/components/ObjectMetadataNavItems.tsx [hunk 1]
```diff
@@ -14,6 +16,9 @@ export const ObjectMetadataNavItems = () => {
   const currentPath = useLocation().pathname;
 
   const { records: views } = usePrefetchedData<View>(PrefetchKey.AllViews);
+  const loading = useIsPrefetchLoading();
+
+  if (loading) return <ObjectMetadataNavItemsSkeletonLoader />;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/components/ObjectMetadataNavItems.tsx:21]: let's avoid if () return and use if () { return }

File: packages/twenty-front/src/modules/favorites/components/Favorites.tsx [hunk 2]
```diff
@@ -32,6 +34,9 @@ const StyledNavigationDrawerItem = styled(NavigationDrawerItem)`
 
 export const Favorites = () => {
   const { favorites, handleReorderFavorite } = useFavorites();
+  const loading = useIsPrefetchLoading();
+
+  if (loading) return <FavoritesSkeletonLoader />;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/favorites/components/Favorites.tsx:39]: same here!

File: packages/twenty-front/src/modules/activities/timeline/components/Timeline.tsx [hunk 3]
```diff
@@ -104,7 +40,7 @@ export const Timeline = ({
   );
 
   if (loading === true) {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/activities/timeline/components/Timeline.tsx:42]: if (loading)

File: packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx [hunk 4]
```diff
@@ -142,7 +122,11 @@ export const RecordDetailRelationSection = ({
 
   const showContent = () => {
     if (loading) {
-      return <StyledRecordDetailRelationSectionSkeletonLoader />;
+      return (
+        <RecordDetailRelationSectionSkeletonLoader
+          isPeopleField={fieldName === 'people'}
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-show/record-detail-section/components/RecordDetailRelationSection.tsx:127]: why do we need to know if People? I feel it's not the right rule, what's the intent?

File: packages/twenty-front/src/modules/support/components/SupportChat.tsx [hunk 5]
```diff
@@ -98,6 +101,8 @@ export const SupportChat = () => {
     currentWorkspaceMember,
   ]);
 
+  if (loading) return <StyledSupportChatSkeletonLoader />;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/support/components/SupportChat.tsx:104]: not sure we need a SkeletonLoader for the support button, do we?

**Review by charlesBochet — APPROVED** [review]
This is a good one, congratulations!

Left a few comments. Could you also add stories on the SkeletonLoaders + one on the whole app in loading mode?

**Comment by gitstart-twenty** [comment]
> This is a good one, congratulations!
> 
> Left a few comments. Could you also add stories on the SkeletonLoaders + one on the whole app in loading mode?

Hey @charlesBochet 
Thanks for the comments! Pushing updates ASAP!

File: packages/twenty-front/src/modules/support/components/SupportChat.tsx (hunk 5 above)
Comment (gitstart-twenty) [reply packages/twenty-front/src/modules/support/components/SupportChat.tsx]: The figma designs indicate so!

File: packages/twenty-front/src/modules/object-metadata/components/ObjectMetadataNavItems.tsx (hunk 1 above)
Comment (gitstart-twenty) [reply packages/twenty-front/src/modules/object-metadata/components/ObjectMetadataNavItems.tsx]: going to check if we can have eslint enfore this

File: packages/twenty-front/src/pages/object-record/__stories__/RecordIndexPage.stories.tsx [hunk 6]
```diff
@@ -44,6 +46,8 @@ export const Default: Story = {
 };
 
 export const Loading: Story = {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
   decorators: [PageDecorator],
```
Comment (gitstart-twenty) [inline packages/twenty-front/src/pages/object-record/__stories__/RecordIndexPage.stories.tsx:51]: For some reason, a decorator with typed `args` will throw an error if placed in a `Story`

**Comment by gitstart-twenty** [comment]
Hey @charlesBochet 
This has been updated!

**Review by charlesBochet — APPROVED** [review]
Thank you!

**Pull request merged**